tensorflow-framework/src/main/java/org/tensorflow/framework/activations/Activation.java

```diff
@@ -14,24 +14,104 @@
 =======================================================================*/
 package org.tensorflow.framework.activations;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
 import org.tensorflow.Operand;
 import org.tensorflow.op.Ops;
 import org.tensorflow.types.family.TNumber;
 
-/**
- * Interface for Activations
- *
- * @param <T> the data type of the input and the result
- */
+/** Interface for Activations */
 @FunctionalInterface
-public interface Activation<T extends TNumber> {
+public interface Activation {
+  Map<String, Supplier<Activation>> createMap =
+      new HashMap<String, Supplier<Activation>>() {
+        {
+          put("elu", ELU::new);
+          put("exponential", Exponential::new);
+          put("gelu", GELU::new);
+          put("hard_sigmoid", HardSigmoid::new);
+          put("linear", Linear::new);
+          put("relu", ReLU::new);
+          put("selu", SELU::new);
+          put("sigmoid", Sigmoid::new);
+          put("softmax", Softmax::new);
+          put("softplus", Softplus::new);
+          put("softsign", Softsign::new);
+          put("swish", Swish::new);
+          put("tanh", Tanh::new);
+        }
+      };
+
+  Map<String, Function<Map<String, Object>, Activation>> createMapConfig =
+      new HashMap<String, Function<Map<String, Object>, Activation>>() {
+        {
+          put("elu", ELU::new);
+          put("exponential", Exponential::new);
+          put("gelu", GELU::new);
+          put("hard_sigmoid", HardSigmoid::new);
+          put("linear", Linear::new);
+          put("relu", ReLU::new);
+          put("selu", SELU::new);
+          put("sigmoid", Sigmoid::new);
+          put("softmax", Softmax::new);
+          put("softplus", Softplus::new);
+          put("softsign", Softsign::new);
+          put("swish", Swish::new);
+          put("tanh", Tanh::new);
+        }
+      };
+
+  /**
+   * Creates an Activation based on a name as known to the TensorFlow engine.
+   *
+   * @param name the activation name
+   * @return the Activation
+   */
+  static Activation create(String name) {
```

Comment:
The doc should note that it defaults to linear if the name is null and throws IAE if the name is unknown.